### PR TITLE
Refactor Transform3 with Matrix Actor

### PR DESCRIPTION
**Pull request by beomki-yeo**
This PR makes `cmath::transform3` have `matrix::actor` as a template parameter.
Hard coded inversion and determinant of 4x4 matrix are moved to cmath::matrix algorithms as well

### CODE REVIEW

**Review by niermann999 — CHANGES_REQUESTED** [review]

**Comment by niermann999** [comment]
Oh, I just saw it is not actually used in the ```vc::transform3```, nvm

File: frontend/vc_vc/include/algebra/vc_vc.hpp [hunk 1]
```diff
@@ -184,7 +191,7 @@ using hard_coded = cmath::matrix::determinant::hard_coded<
 // preset(s) as standard option(s) for user's convenience
 template <typename size_type, typename scalar_t>
 using preset0 = actor<size_type, scalar_t, cofactor<size_type, scalar_t>,
-                      hard_coded<size_type, scalar_t, 2>>;
+                      hard_coded<size_type, scalar_t, 2, 4>>;
```
[the remark this answers is not in the document]
Comment (niermann999) [reply frontend/vc_vc/include/algebra/vc_vc.hpp:194]: Oh, I just saw it is not actually used in the vc::transform3, nvm

File: frontend/vc_cmath/include/algebra/vc_cmath.hpp [hunk 2]
```diff
@@ -109,19 +81,26 @@ using vc::math::normalize;
...
 
+template <typename T, std::size_t N>
+using array_type = Vc::array<T, N>;
+
 template <typename T, std::size_t ROWS, std::size_t COLS>
 using matrix_type = vc::matrix_type<T, ROWS, COLS>;
 
 template <typename size_type, typename scalar_t>
 using element_getter_type =
     cmath::element_getter<size_type, Vc::array, scalar_t>;
```
[the remark this answers is not in the document]
Comment (beomki-yeo) [reply frontend/vc_cmath/include/algebra/vc_cmath.hpp]: Nice catch. just fixed it

**Comment by beomki-yeo** [comment]
Please don't mind that I merge this without an approval ;-) feel like to update detray today with new version of algebra-plugins

**Pull request merged**